website/common/script/ops/changeClass.js

```diff
@@ -47,32 +71,19 @@ module.exports = function changeClass (user, req = {}, analytics) {
 
     if (analytics) {
       analytics.track('change class', {
-        uuid: user._id,
-        class: klass,
-        acquireMethod: 'Gems',
-        gemCost: 3,
-        category: 'behavior',
-        headers: req.headers,
+          uuid: user._id,
+          class: klass,
+          acquireMethod: 'Gems',
+          gemCost: 3,
+          category: 'behavior',
+          headers: req.headers,
       });
     }
   } else {
-    if (user.preferences.disableClasses) {
-      user.preferences.disableClasses = false;
-      user.preferences.autoAllocate = false;
-    } else {
-      if (user.balance < 0.75) throw new NotAuthorized(i18n.t('notEnoughGems', req.language));
-      user.balance -= 0.75;
-    }
-
-    user.stats.str = 0;
-    user.stats.con = 0;
-    user.stats.per = 0;
-    user.stats.int = 0;
-    user.stats.points = capByLevel(user.stats.lvl);
-    user.flags.classSelected = false;
+        // if invalid class is specified, throw an error.
+    throw new BadRequest(i18n.t('invalidClass', req.language));
   }
-
   return [
-    pick(user, splitWhitespace('stats flags items preferences')),
+        pick(user, splitWhitespace('stats flags items preferences')),
```

Comment:
I just fixed up your requests and I am running the tests again now :) Do I make a new pull request once I'm done or is there some way to update this one with my new commit?